pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -186,9 +219,54 @@ func TestHTTPScrapeClient_Scrape_ErrorCases(t *testing.T) {
 		expectedErr:     "could not find value for queue_proxied_operations_per_second in response",
 	}}
 
+	for i, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			for j, hClient := range []*http.Client{
+				newTestHTTPClient(getHTTPResponse(test.responseCode, test.responseContext), test.responseErr),
+				newTestHTTPClient(getProtoHTTPResponse(test.responseCode, emptyStat, "application/protobuf"), test.responseErr),
+			} {
+				// only the first two cases apply for the proto based response, unlike Prometheus gathered statistics
+				// proto will not have any missing values.
+				if i >= 2 && j == 1 {
```

Comment:
For proto metrics there is no scenario where metrics does not exist. The is only the case for the 
Prometheus client library where it does a [best effort](https://github.com/prometheus/client_golang/blob/b0cdec211b8bf9d5b1edabff4d32088a11f18310/prometheus/registry.go#L150) in case of an error.

Reply:
Hmm, let's add an explicit `bool` or something to indicate these shouldN't run on protobuf code or something. This feels super cryptic.

Or just copy & paste only the relevant cases to the test below?